strawberry/schema/types/object_type.py

```diff
@@ -12,32 +12,43 @@ def _get_object_type_for_type_definition(
     type_definition: TypeDefinition, type_map: TypeMap
 ) -> GraphQLType:
 
-    TypeClass: Type = GraphQLObjectType
-
-    kwargs = {}
-
-    if type_definition.is_input:
-        TypeClass = GraphQLInputObjectType
-    elif type_definition.is_interface:
-        TypeClass = GraphQLInterfaceType
-
-    if type_definition.interfaces:
-        kwargs["interfaces"] = [
-            _get_object_type_for_type_definition(interface, type_map)
-            for interface in type_definition.interfaces
-        ]
-
-    assert not type_definition.is_generic
-
-    return TypeClass(
-        name=type_definition.name,
-        fields=lambda: {
-            field.name: get_field(field, type_definition.is_input, type_map)
-            for field in type_definition.fields
-        },
-        description=type_definition.description,
-        **kwargs,
-    )
+    if type_definition.name not in type_map:
```

Comment:
I wonder what would happen now if i define two classes but different with different fields. Should we issue a warning that the duplicate type will be ignored?

Reply:
yes, good point, we should add a check for that :)

https://github.com/strawberry-graphql/strawberry/issues/404